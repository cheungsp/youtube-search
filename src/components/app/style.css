body {
  background-color: whitesmoke;
}

.video-item img {
  max-width: 64px;
}

.video-detail .details {
  margin-top: 10px;
  padding: 10px;
  border: 1px solid #ddd;
  border-radius: 4px;
}