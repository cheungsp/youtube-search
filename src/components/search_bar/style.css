.search-bar {
  margin: 20px;
  text-align: center;
}

.search-bar input {
  width: 80%;
}

/*
"test": "jest"
"test": "react-scripts test --env=jsdom",
*/