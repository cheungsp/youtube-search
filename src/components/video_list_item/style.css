.single-video-item {
  margin-bottom: 15px;
}