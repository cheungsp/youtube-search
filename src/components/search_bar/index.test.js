import _ from 'lodash';
import React from 'react';
import { shallow } from 'enzyme'
import SearchBar from './index.js'
import toJson from 'enzyme-to-json'
import './style.css';


test('Search component should render as expected', () => {
  const component = shallow(<SearchBar />)
  const tree =  toJson(component)
  expect(tree).toMatchSnapshot()
})

it('should pass a selected value to the onChange function', () => {
  const videoSearch = _.debounce((term) => { this.videoSearch(term) }, 200);
    const component = shallow(<SearchBar onSearchTermChange={videoSearch}/>);
    component.find('input').simulate('change', { target: {
      value: 'Change function' }
    });
    expect(toJson(component)).toMatchSnapshot();
});