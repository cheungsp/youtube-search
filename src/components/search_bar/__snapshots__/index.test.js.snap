// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render as expected 1`] = `
<div
  className="search-bar"
>
  <input
    onChange={[Function]}
    placeholder="Search"
    value=""
  />
</div>
`;
 